server/chat.ts

```diff
@@ -92,7 +92,7 @@ export type ChatFilter = ((
 	connection: Connection,
 	targetUser: User | null,
 	originalMessage: string
-) => string | false | null | undefined) & {priority?: number};
+) => string | false | null | undefined | void) & {priority?: number};
```

Comment:
Oh! That's why you had `return undefined`. I think the `return undefined` opt-in might be better, then.